hnn_core/network.py

```diff
@@ -414,6 +416,34 @@ def __repr__(self):
                  len(self.pos_dict['L5_basket'])))
         return '<%s | %s>' % (class_name, s)
 
+    def __eq__(self, other):
+        if not isinstance(other, Network):
+            return NotImplemented
+
+        # Check connectivity
+        if not (len(self.connectivity) == len(other.connectivity)):
+            return False
+        # order of connections doesn't matter
+        for conn in self.connectivity:
+            src_gids = list(conn['src_gids'])
+            target_gids = list(conn['target_gids'])
+            match_conns = pick_connection(other,
+                                          src_gids=src_gids,
+                                          target_gids=target_gids,
+                                          loc=conn['loc'],
+                                          receptor=conn['receptor'])
+            if len(match_conns) == 0:
+                return False
+
+        # Check all other attributes
+        for attr in ['cell_types', 'gid_ranges', 'pos_dict', 'cell_response',
```

Comment:
Comment for everyone, @jasmainak @rythorpe @raj1701 

Any way we can add a safe guard to make sure the equality operation checks any new attributes added to the network? Ideally something that breaks the code when you add a public attribute not reflected in this list.

Reply:
We could potentially use `self.__dict__` which should return a dict of all attribute names and their respective values for the given object instance.

Reply:
Or the [`dir()`](https://docs.python.org/3/library/functions.html#dir) built-in....

Reply:
Sure will make this change.